sklearn/linear_model/_base.py

```diff
@@ -518,7 +527,29 @@ def fit(self, X, y, sample_weight=None):
             # Sample weight can be implemented via a simple rescaling.
             X, y = _rescale_data(X, y, sample_weight)
 
-        if sp.issparse(X):
+        if self.solver == "cholesky":
+            n_samples, n_features = X.shape
+            ravel = False
+            if y.ndim == 1:
+                y = y.reshape(-1, 1)
+                ravel = True
+            n_samples_, n_targets = y.shape
+            alpha = np.asarray(0, dtype=X.dtype).ravel()
```

Comment:
Suggestion:
alpha = np.array([0], dtype=X.dtype)

Reply:
Thanks! Changed to this.